diff --git a/include/assimp/defs.h b/include/assimp/defs.h
index 20b234f9..85a66737 100644
--- a/include/assimp/defs.h
+++ b/include/assimp/defs.h
@@ -285,6 +285,6 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  * This can be adjusted for specific use cases
  * It's NOT a total limit, just a limit for individual allocations
  */
-#define AI_MAX_ALLOC(type) ((256U * 1024 * 1024) / sizeof(type))
+#define AI_MAX_ALLOC(type) ((size_t(256) * 1024 * 1024 * 1024) / sizeof(type))
 
 #endif // !! AI_DEFINES_H_INC
